libsolidity/formal/SolverInterface.h

```diff
@@ -42,21 +42,65 @@ enum class CheckResult
 	SATISFIABLE, UNSATISFIABLE, UNKNOWN, CONFLICTING, ERROR
 };
 
-enum class Sort
+enum class Kind
 {
 	Int,
-	Bool
+	Bool,
+	Array,
+	Function
+};
+
+struct Sort
+{
+	Sort(Kind _kind):
+		kind(_kind) {}
+	virtual ~Sort() = default;
+	Kind const kind;
+	bool operator==(Sort const& _other) const { return kind == _other.kind; }
+};
+using SortPointer = std::shared_ptr<Sort>;
```

Comment:
I think you can make a refactoring PR which only introduces `struct Sort`.

Reply:
Created #5466